src/nativeMain/kotlin/com/autodesk/coroutineworker/BackgroundCoroutineWorkQueueExecutor.kt

```diff
@@ -39,38 +38,58 @@ internal interface CoroutineWorkItem {
 internal class BackgroundCoroutineWorkQueueExecutor<WorkItem : CoroutineWorkItem>(private val numWorkers: Int) {
 
     /**
-     * Locks access during enqueue and dequeue operations
-     * to ensure counts can be consistently read across threads
-     * */
-    private val queueLock = Lock()
+     * The pool, on which blocks are executed
+     */
+    private val pool = WorkerPool(numWorkers)
 
-    /** The queue of WorkItems to execute */
-    private val queue = frozenLinkedList<WorkItem>()
+    /**
+     * Protects access to the queue on a single thread
+     */
+    private val queueThread = Worker.start()
 
-    /** The pool, on which blocks are executed */
-    private val pool = WorkerPool(numWorkers)
+    /**
+     * The wrapped (allow freezing and mutable access on single thread) queue of WorkItems
+     */
+    private val wrappedQueue = AtomicReference<StableRef<WorkQueue<WorkItem>>?>(null)
```

Comment:
@kpgalligan is there some other way to do this? I have a queue that I only mutate from a single thread, but it needs to live in this frozen object

Reply:
improved slightly with `by lazy`